adapter/comment_adapter.go

```diff
@@ -0,0 +1,123 @@
+package adapter
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/nibble-4bits/ondemand-go-bootcamp/entity"
+)
+
+var (
+	ErrCommentNotFoundByID = errors.New("no comment found by ID")
+)
+
+type commentAdapter struct {
+	csvDataSource  CSVDataSource
+	httpDataSource HTTPDataSource
+	dataStore      DataStore
+	comments       []entity.Comment
+}
+
+// NewCommentAdapter receives a data source and will try to fetch the
```

Comment:
For convention. In go you document functions with the functions name and description like:

// calculateSum - This function calculate the sum of 2 numbers